src/AkvoDjangoFormGateway/models.py

```diff
@@ -1,3 +1,114 @@
-# from django.db import models
+import uuid
+from django.db import models
+from .constants import QuestionTypes
+
 
 # Create your models here.
+class AkvoGatewayForm(models.Model):
+    name = models.CharField(max_length=255)
+    description = models.TextField(null=True, default=None)
+    version = models.IntegerField(default=1)
+    uuid = models.UUIDField(default=uuid.uuid4, editable=False, unique=True)
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        db_table = 'ag_form'
+
+
+class AkvoGatewayQuestion(models.Model):
+    form = models.ForeignKey(
+        to=AkvoGatewayForm,
+        on_delete=models.CASCADE,
+        related_name='ag_form_questions',
+    )
+    order = models.BigIntegerField(null=True, default=None)
+    text = models.TextField()
+    type = models.IntegerField(choices=QuestionTypes.FieldStr.items())
+    required = models.BooleanField(default=True)
+
+    def __str__(self):
+        return self.text
+
+    def to_definition(self):
+        options = (
+            [
+                options.name
+                for options in self.ag_question_question_options.all()
+            ]
+            if self.ag_question_question_options.count()
+            else False
+        )
+        return {
+            'id': self.id,
+            'order': self.order + 1,
+            'text': self.text,
+            'type': QuestionTypes.FieldStr.get(self.type),
+            'required': self.required,
+            'options': options,
+        }
+
+    class Meta:
+        db_table = 'ag_question'
+
+
+class AkvoGatewayQuestionOption(models.Model):
+    question = models.ForeignKey(
+        to=AkvoGatewayQuestion,
+        on_delete=models.CASCADE,
+        related_name='ag_question_question_options',
+    )
+    order = models.BigIntegerField(null=True, default=None)
+    code = models.CharField(max_length=255, default=None, null=True)
+    name = models.TextField()
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        db_table = 'ag_option'
+
+
+class AkvoGatewayData(models.Model):
+    name = models.TextField()
+    form = models.ForeignKey(
+        to=AkvoGatewayForm,
+        on_delete=models.CASCADE,
+        related_name='ag_form_data',
+    )
+    geo = models.JSONField(null=True, default=None)
+    phone = models.CharField(max_length=25)
+    created = models.DateTimeField(auto_now_add=True)
+    updated = models.DateTimeField(default=None, null=True)
```

Comment:
where is the status? Submitted boolean?

Reply:
Sorry, I missed adding this prop. What if the submitted field is changed to "completed" as a boolean?
completed = True (All questions answered and status Completed)
completed = False (Only a few questions answered and status is Draft)

Reply:
I think instead of boolean, we can have an integer field with descriptive choices (e.g., 1 for draft, 2 for submitted) so the code more readable and self-explanatory